### PR TITLE
[GS2-282] Implement user reservations metadata endpoint

### PR DESCRIPTION
**Pull request by VitaliiChernenko**
Task: [282](https://team-gs2.atlassian.net/browse/GS2-282?atlOrigin=eyJpIjoiMzRjNjUzNmFkYmUwNGZjN2FkY2UyZWRmZjgzNTdkM2QiLCJwIjoiaiJ9)

## Summary by CodeRabbit

* **New Features**
  * Added an endpoint to fetch your reservation metadata: remaining money, remaining items, and a list of reserved products with reservation timestamps.
  * Reservation limits are now configurable via application settings.

* **Tests**
  * Added unit, integration, and end-to-end tests covering metadata retrieval, mapping, and boundary cases for limits.

### CODE REVIEW

File: code/orders-api-rest/api/openapi-rest.yml [hunk 1]
```diff
@@ -2854,6 +2875,43 @@ components:
...
       description: Status of the product
 
+    UserReservationsMetadata:
+      type: object
+      description: Metadata about the user's reserved products.
+      properties:
+        remainingMoney:
+          type: number
+          format: double
+          example: 899.25
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest/api/openapi-rest.yml]: Nice catch.
Thanks, fixed.

File: code/orders-api-rest/api/openapi-rest.yml [hunk 2]
```diff
@@ -2854,6 +2875,43 @@ components:
...
+      properties:
+        id:
+          type: string
+          format: uuid
+          example: b8be6872-b25e-4c1a-b845-df2f0150b4f2
+          description: Identifier of the reserved product.
+        reservedAt:
+          type: string
+          format: date-type
+          example: '2026-02-24T13:42:38.196466'
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest/api/openapi-rest.yml:2909]: You are right.
Thanks, fixed.

File: code/orders-infrastructure/src/main/java/com/academy/orders/infrastructure/reservation/repository/ReservationsRepositoryImpl.java [hunk 3]
```diff
@@ -50,6 +51,13 @@ public List<Product> getReservationProducts(Long accountId, String language) {
         .toList();
   }
 
+  @Override
+  public List<ReservationMetadata> getUserReservationMetadata(Long accountId) {
+    return reservationsJpa.findByIdUserId(accountId).stream()
+        .map(entity -> new ReservationMetadata(entity.getId().getProductId(), entity.getAddedAt()))
+        .toList();
+  }
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-infrastructure/src/main/java/com/academy/orders/infrastructure/reservation/repository/ReservationsRepositoryImpl.java:60]: Yeah, makes sense.
Thanks, fixed.

**Review by vitkikh — APPROVED** [review]
